### PR TITLE
Setup front web socket connection and use it to display waiting room player list.

### PR DESCRIPTION
**Pull request by Scrabouillmax**

### CODE REVIEW

File: backend/src/gameRoomManager.ts [hunk 1]
```diff
@@ -17,20 +18,31 @@ export class GameRoomManager {
     };
   };
 
+  getPlayers = (): string[] => Object.keys(this.playersConnections);
```
Comment (Scrabouillmax) [inline backend/src/gameRoomManager.ts]: For now I use this, let's add pseudos later on

File: frontend/src/websockets/ConnectionManager.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+   */
+  public registerCallbackForMessage = (
+    type: MessageType,
+    callback: MessageCallback
+  ): void => {
+    this.messageCallbacks[type] = [...this.messageCallbacks[type], callback];
+  };
+
+  private ping = (): void => {
+    console.log("ping");
```
Comment (2016floreza) [inline frontend/src/websockets/ConnectionManager.ts]: Is this still wanted ?

Comment (Scrabouillmax) [reply frontend/src/websockets/ConnectionManager.ts]: I find that useful to debug the connection but we can remove for now

File: backend/src/controllers/websockets_controller.ts [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 
-enum MessageType {
+export enum InboundMessageType {
   PING = "PING",
   JOIN_GAME = "JOIN_GAME",
+}
+
+export enum OutboundMessageType {
+  WAITING_ROOM_UPDATE = "WAITING_ROOM_UPDATE",
   SUCCESS = "SUCCESS",
```
Comment (quentinverlhac) [inline backend/src/controllers/websockets_controller.ts]: Should this be removed ? It doesn't seemed to be used anymore.
```suggestion
```

File: frontend/src/websockets/MessageType.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+enum MessageType {
+  PING = "PING",
+  JOIN_GAME = "JOIN_GAME",
+  WAITING_ROOM_UPDATE = "WAITING_ROOM_UPDATE"
+}
```
Comment (quentinverlhac) [inline frontend/src/websockets/MessageType.ts]: What about `ERROR` messages ?

File: backend/src/controllers/websockets_controller.ts (hunk 3 above)
Comment (Scrabouillmax) [reply backend/src/controllers/websockets_controller.ts]: Indeed :) I commit the suggestion

File: frontend/src/websockets/MessageType.ts (hunk 4 above)
Comment (Scrabouillmax) [reply frontend/src/websockets/MessageType.ts]: We need to think deeper I about error handling, I suggest we do a separate PR with a fully-fledged error handling (i.e. either error page, retry mecanism...).

**Pull request merged**